oracle/blockeligibilityvalidator.go

```diff
@@ -74,21 +70,21 @@ func (v BlockEligibilityValidator) BlockEligible(block *types.BlockHeader) (bool
 	return block.LayerIndex == eligibleLayer, nil
 }
 
-func (v BlockEligibilityValidator) getValidATX(blockEpoch types.EpochId, block *types.BlockHeader) (*types.ActivationTx, error) {
+func (v BlockEligibilityValidator) getActiveSetSize(block *types.BlockHeader) (uint32, error) {
+	blockEpoch := block.LayerIndex.GetEpoch(v.layersPerEpoch)
+	if blockEpoch.IsGenesis() {
+		return v.committeeSize, nil
+	}
 	atx, err := v.activationDb.GetAtx(block.ATXID)
 	if err != nil {
 		v.log.Error("getting ATX failed: %v %v ep(%v)", err, block.ATXID.ShortString(), blockEpoch)
-		return nil, fmt.Errorf("getting ATX failed: %v %v ep(%v)", err, block.ATXID.ShortString(), blockEpoch)
-	}
-	if !atx.Valid {
-		v.log.Error("ATX %v is invalid", atx.ShortId())
-		return nil, fmt.Errorf("ATX %v is invalid", atx.ShortId())
+		return 0, fmt.Errorf("getting ATX failed: %v %v ep(%v)", err, block.ATXID.ShortString(), blockEpoch)
 	}
 	if atxTargetEpoch := atx.PubLayerIdx.GetEpoch(v.layersPerEpoch) + 1; atxTargetEpoch != blockEpoch {
 		v.log.Error("ATX target epoch (%d) doesn't match block publication epoch (%d)",
 			atxTargetEpoch, blockEpoch)
-		return nil, fmt.Errorf("ATX target epoch (%d) doesn't match block publication epoch (%d)",
+		return 0, fmt.Errorf("ATX target epoch (%d) doesn't match block publication epoch (%d)",
```

Comment:
you can remove this entire `if`, it is tested before this method is called

Reply:
added todo